src/server/index.ts

```diff
@@ -131,16 +131,80 @@ export class DocumentDriveServer extends BaseDocumentDriveServer {
         this.initializePromise = this._initialize();
     }
 
-    private updateSyncStatus(
-        driveId: string,
-        status: SyncStatus | null,
+    private getOperationSource(source: StrandUpdateSource) {
+        return source.type === 'local' ? 'push' : 'pull';
+    }
+
+    private getCombinedSyncUnitStatus(
+        syncUnitStatus: SyncUnitStatusObject
+    ): SyncUnitStatus {
+        if (syncUnitStatus.pull === 'INITIAL_SYNC') return 'INITIAL_SYNC';
+        if (syncUnitStatus.push === 'INITIAL_SYNC') return syncUnitStatus.pull;
+
+        const order = ['ERROR', 'MISSING', 'CONFLICT', 'SYNCING', 'SUCCESS'];
+        const sortedStatus = Object.values(syncUnitStatus).sort(
+            (a, b) => order.indexOf(a) - order.indexOf(b)
+        );
+
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        return sortedStatus[0]!;
+    }
+
+    private updateSyncUnitStatus(
+        syncUnitId: string,
+        status: Partial<SyncUnitStatusObject> | null,
         error?: Error
     ) {
         if (status === null) {
-            this.syncStatus.delete(driveId);
-        } else if (this.syncStatus.get(driveId) !== status) {
-            this.syncStatus.set(driveId, status);
-            this.emit('syncStatus', driveId, status, error);
+            this.syncStatus.delete(syncUnitId);
+            return;
+        }
+
+        const syncUnitStatus = this.syncStatus.get(syncUnitId);
+
+        if (!syncUnitStatus) {
+            const defaultSyncUnitStatus: SyncUnitStatusObject = {
+                pull:
+                    status.pull && status.pull !== 'SYNCING'
```

Comment:
if we  try to update the sync unit to `SYNCING`, but there's no previous state, then we use 'INITIAL_SYNC' instead